solidity/contracts/converter/types/liquid-token/LiquidTokenConverter.sol

```diff
@@ -249,24 +251,24 @@ contract LiquidTokenConverter is ConverterBase {
         IERC20Token reserveToken = reserveTokens[0];
 
         // ensure that the trade will only deplete the reserve balance if the total supply is depleted as well
-        uint256 tokenSupply = ISmartToken(anchor).totalSupply();
+        uint256 tokenSupply = ISmartToken(address(anchor)).totalSupply();
         uint256 rsvBalance = reserveBalance(reserveToken);
         assert(amount < rsvBalance || (amount == rsvBalance && _amount == tokenSupply));
 
         // destroy the tokens from the converter balance in the liquid token
-        ISmartToken(anchor).destroy(this, _amount);
+        ISmartToken(address(anchor)).destroy(address(this), _amount);
 
         // update the reserve balance
         reserves[reserveToken].balance = reserves[reserveToken].balance.sub(amount);
 
         // transfer funds to the beneficiary in the reserve token
-        if (reserveToken == ETH_RESERVE_ADDRESS)
+        if (address(reserveToken) == ETH_RESERVE_ADDRESS)

```

Comment:
If we use `ETH_RESERVE_ADDRESS` as `IERC20Token` everywhere in this file, then you may want to consider declaring it as such to begin with.